src/main/java/org/example/Main.java

```diff
@@ -1,21 +1,74 @@
 package org.example;
 
-import org.example.spaces.CreaturesFactory;
-import org.example.spaces.PetShopFactory;
-import org.example.spaces.ZooFactory;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Main {
     public static void main(String[] args) {
-        Application application = configureApplication(args[0]);
-        application.checkCreaturesNames();
-        application.areCreaturesSafe();
+        List<Integer> sortedList = getSortedList(100);
+        int numberToSearch = 49;
+
+        int myResult = beforeGoogleBinarySearch(sortedList, numberToSearch);
+
+        int goodResult = afterGoogleBinarySearch(sortedList, numberToSearch);
+
+        
     }
 
-    private static Application configureApplication(final String animalDangerDescription) {
-        return new Application(getProperFactory(animalDangerDescription));
+    //this is an example from the internet - it works fine
+    private static int afterGoogleBinarySearch(List<Integer> sortedList, int valueToFind) {
+        int index = -1;
+        int low = 0;
+        int high = sortedList.size();
+
+        while (low <= high) {
+            int mid = low + (high - low) / 2;
+            if (sortedList.get(mid) < valueToFind) {
+                low = mid + 1;
+            } else {
+                if (sortedList.get(mid) > valueToFind) {
+                    high = mid - 1;
+                } else {
+                    if (sortedList.get(mid) == valueToFind) {
+                        index = mid;
+                        break;
+                    }
+                }
+            }
+        }
+
+        return index;
+    }
+
+    //this is my personal solution - it also works, but while(true) is really bad - i didn't know how to stop cycle in while(...)
+    private static int beforeGoogleBinarySearch(List<Integer> sortedList, int numberToSearch) {
+        int result;
+        int start = 0;
+        int end = sortedList.size();
+
+        while (true) {
```

Comment:
To avoid infinite loop
```suggestion
        while (start <= end) {
```